cli/serve/serve.go

```diff
@@ -244,9 +248,55 @@ func serverMain(args []string, c cli.Config) error {
 		log.Info("Now listening on ", addr)
 		return http.ListenAndServe(addr, nil)
 	}
+	if conf.MutualTLSCAFile != "" {
+		mutualTLSCAs, err := ioutil.ReadFile(conf.MutualTLSCAFile)
+		if err != nil {
+			return fmt.Errorf("Failed to load mutual TLS CA file: %s", err)
+		}
+		cert, err := ioutil.ReadFile(conf.TLSCertFile)
+		if err != nil {
+			return fmt.Errorf("Failed to load cert file: %s", err)
+		}
+		key, err := ioutil.ReadFile(conf.TLSKeyFile)
+		if err != nil {
+			return fmt.Errorf("Failed to load cert file: %s", err)
+		}
+		keypair, err := tls.X509KeyPair(cert, key)
```

Comment:
You don't need to save the `cert`, `key`, or `keypair` here (or manually set `Certificates` in the `TLSConfig`. http.ListenAndServeTLS will [do that for you](https://golang.org/src/net/http/server.go?s=60146:60214#L2026)! Also, should the need arise there is a `tls.LoadX509KeyPair` helper that will read the files for you :)